### ISSUE TITLE
Feat: base16-vim integration

### ISSUE THREAD

**Issue by jonleopard** [nvim-lualine/lualine.nvim#859]
### Requested feature

[base16-vim](https://github.com/base16-project/base16-vim) and [base16-shell](https://github.com/base16-project/base16-shell) offer a suite of tools that allow automatic colorscheme switching in multiple environments (shell, tmux, vim, etc..). With `base16-vim` installed, neovim is aware of what theme is currently active and will set the theme accordingly.

Is there a way that we could use `base16-vim` instead of [nvim-base16](https://github.com/RRethy/nvim-base16)?

Ideally, it would be nice to tell `lualine` to use whatever `base16` theme is currently active. This also opens up automatic theme switching for users that have `base16-shell` installed.

```
local current_theme_name = vim.g.colors_name

require 'lualine'.setup {
    options = {
        theme = current_theme_name
    }
}
```

Right now, this returns an error since its not aware of `base16-vim`'s colorschemes.

```
Theme `base16-tokyo-night-dark` not found, falling back to `auto`. Check if spelling is right.

theme(base16): nvim-base16 is not currently present in your runtimepath, make sure it is properly installed, fallback to default colors.
```

### Motivation

The base16 project is fairly centralized and themes are updated weekly. Instead of relying on another repository (`nvim-base16`) which might have out of date themes and no integration with the core suite of tools, I think integrating lualine to work with `base16-vim` would be cool/useful idea.

Thanks!

**Comment by shadmansaleh** [comment]
It's possible to support base16-vim too. Some one just needs to import the colors from base16-vim when it's installed [here](https://github.com/nvim-lualine/lualine.nvim/blob/edca2b03c724f22bdc310eee1587b1523f31ec7c/lua/lualine/themes/base16.lua#L77) And update the base16 conditions to check for base16-vim alongside base nvm-base16

**Comment by t3hpr1m3** [comment]
Slight necro here, but the issue is that [base16-vim](https://github.com/base16-project/base16-vim) doesn't actually "load" any colors.  It just sets `colorscheme` normally, and `rtp` logic loads the colors file. In essence, the correct behavior is just to use `auto` as the theme.  Unfortunately, because the `colorscheme` contains `base16` in the name, it triggers the [nvim-base16](https://github.com/RRethy/nvim-base16) integration, which results in a `LualineNotice` because the plugin isn't loaded.

In looking at the way the logic is laid out in [lualine.nvim](https://github.com/nvim-lualine/lualine.nvim), I'm not sure what the best fix would be, since the assumption is that if you're using a `base16` theme, you have [nvim-base16](https://github.com/RRethy/nvim-base16) installed (or at least something that behaves like it).

**Comment by titaniumtraveler** [comment]
@t3hpr1m3 Wouldn't the best option not be to just test if `vim.g.base16_gui00` etc are set and if yes to just use that to create the `colors` table?
That way it should just work automatically.

**Comment by t3hpr1m3** [comment]
Sorry for the huge delay.  That would work (and your PR looks like it would work).  For transparency, this issue caused me to drop `lualine` and just put together a quick custom statusline, so somebody else might be able to verify.

**Comment by shadmansaleh** [comment]
resolved with #1006
